### PR TITLE
MB-9461-stop-prime-sit-extension

### PR DESCRIPTION
**Pull request by NamibiaTorres**
## Description

As the prime

I want to be prevented from submitting a SIT Extension request if one is already pending

So that the TOO doesn’t get overwhelmed with requests

---
Additionally, as part of this ticket tests for the required fields check were added because they were previously missing.

## Setup
To run tests: 

```go
go test ./pkg/services/sit_extension --testify.m TestValidationRules -v
```

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-9461) for this change

### CODE REVIEW

**Review by sandy-wright — COMMENTED** [review]
I have one small comment, but this looks good! @NamibiaTorres, would you mind merging in master so I can try an e2e test? Looks like this branch doesn't include the `createSITExtension` endpoint

**Review by sandy-wright — APPROVED** [review]
I went ahead and merged master in locally so I could test it. Worked well! Thanks for adding all of those test cases too.

File: pkg/services/sit_extension/rules_test.go [hunk 1]
```diff
@@ -32,6 +32,129 @@ func (suite *SitExtensionServiceSuite) TestValidationRules() {
...
+					RequestedDays: 90,
+				},
+			})
+
+			err := checkRequiredFields().Validate(appcontext.NewAppContext(suite.DB(), suite.logger), sitExtension, nil)
+			switch verr := err.(type) {
+			case *validate.Errors:
+				suite.NoVerrs(verr)
+			default:
+				suite.Failf("expected *validate.Errs", "%v", err)
```
[the remark this answers is not in the document]
Comment (NamibiaTorres) [reply pkg/services/sit_extension/rules_test.go:57]: Yes!

**Comment by NamibiaTorres** [comment]
> I went ahead and merged master in locally so I could test it. Worked well! Thanks for adding all of those test cases too.
No problem! I also merged master into my branch.

**Pull request merged**